databases/week1/sqlhomeworkweek1.sql

```diff
@@ -0,0 +1,47 @@
+-- 1. Find out how many tasks are in the task table
+SELECT COUNT(id) AS task FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id) FROM task 
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT * FROM task WHERE status_id = 3;
+
+-- 4. Find all the tasks that are not marked as done
+SELECT * FROM task WHERE status_id = 0;
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT * FROM task 
+ORDER BY created DESC;
+
+-- 6. Get the single most recently created task
+SELECT * FROM task 
+ORDER BY created DESC
+LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date FROM task 
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+--  8. Get the title and status (as text) of all tasks
+SELECT t.title, s.name 
+FROM task t
+JOIN status s
+ON t.status_id = s.id; 
+
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT s.name, COUNT(*) AS 'number of tasks'  
+FROM status s 
+LEFT JOIN task t
+ON t.status_id = s.id
+GROUP BY s.id; 
```

Comment:
Nice. I would avoid putting spaces in the alias though. Instead i would call it something like `number_of_tasks`